index_test.go

```diff
@@ -1076,23 +1078,47 @@ func TestNegativeRepo(t *testing.T) {
 }
 
 func TestListRepos(t *testing.T) {
-	content := []byte("bla the needle")
+	content := []byte("bla the needle\n")
 	// ----------------01234567890123
-	b := testIndexBuilder(t, &Repository{
-		Name: "reponame",
-	},
-		Document{Name: "f1", Content: content},
-		Document{Name: "f2", Content: content})
+	repo := &Repository{
+		Name:     "reponame",
+		Branches: []RepositoryBranch{{Name: "main"}, {Name: "dev"}},
+	}
+	b := testIndexBuilder(t, repo,
+		Document{Name: "f1", Content: content, Branches: []string{"main", "dev"}},
+		Document{Name: "f2", Content: content, Branches: []string{"main"}},
+		Document{Name: "f2", Content: content, Branches: []string{"dev"}},
+		Document{Name: "f3", Content: content, Branches: []string{"dev"}})
 
 	searcher := searcherForTest(t, b)
 	q := &query.Repo{Pattern: "epo"}
 	res, err := searcher.List(context.Background(), q)
 	if err != nil {
 		t.Fatalf("List(%v): %v", q, err)
 	}
-	if len(res.Repos) != 1 || res.Repos[0].Repository.Name != "reponame" {
-		t.Fatalf("got %v, want 1 matches", res)
+
+	want := &RepoList{
+		Repos: []*RepoListEntry{{
+			Repository: *repo,
+			Stats: RepoStats{
+				Documents:    4,
+				ContentBytes: 68,
```

Comment:
Do the 68 bytes include the size of the slice headers?

Reply:
nope. filenames are part of content though count. So each content is 15 bytes, and each filename is 2 bytes in the test. `(15 + 2) * 4 == 68`